tiledb/api/src/convert.rs

```diff
@@ -29,3 +29,52 @@ impl<T: CAPISameRepr> CAPIConverter for T {
         *value
     }
 }
+
+/// Trait for comparisons based on value bits.
+/// This exists to work around float NaN which is not equal to itself,
```

Comment:
I would love, love, love if we did not need this.

Fun fact:
`impl<T> BitsEq for T where T: Eq` is forbidden in concert with `impl BitsEq for f32` because the compiler says that `std` may impl Eq for f32 someday.  No it won't.

Reply:
i'd actually add this as a comment before merging. Without having had the context of knowing there was a NaN issue to begin with, this particular change without context might be a bit odd. And in the crazy case they ever do implement `Eq` for `f32` we'll have the confidence to remove it.